### PR TITLE
core: Implement first cases of assembly format optional groups

### PR DESCRIPTION
**Pull request by PapyChacal**
This covers optional groups with no else group, only literal first then element, and optional/variadic operand/result/type anchors. I think this is the most widely used, it enables the typical `` (`(` $variadic_operand`)`)?``

### CODE REVIEW

File: tests/irdl/test_declarative_assembly_format.py [hunk 1]
```diff
@@ -1143,3 +1143,73 @@ class VarOpGuardOp(IRDLOperation):  # pyright: ignore[reportUnusedClass]
...
+@pytest.mark.parametrize(
+    "program, generic_program",
+    [
+        (
+            '%0 = "test.op"() : () -> i32\n' "test.optional_group(%0 : i32)",
+            '%0 = "test.op"() : () -> i32\n' '"test.optional_group"(%0) : (i32) -> ()',
+        ),
+        (
+            "test.optional_group",
+            '"test.optional_group"() : () -> ()',
```
Comment (superlopuh) [inline tests/irdl/test_declarative_assembly_format.py:1158]: can you please add another test case with more operands?

**Review by superlopuh — APPROVED** [review]

File: xdsl/transforms/experimental/convert_stencil_to_linalg.py
Comment (superlopuh) [inline xdsl/transforms/experimental/convert_stencil_to_linalg.py:1]: ?

Comment (PapyChacal) [reply xdsl/transforms/experimental/convert_stencil_to_linalg.py:1]: Oh no :smiling_face_with_tear:

Comment (PapyChacal) [reply xdsl/transforms/experimental/convert_stencil_to_linalg.py:1]: Oh good, just a spare empty file, will revert and merge

**Pull request merged**